### PR TITLE
armsr: enable more features / hardware

### PR DESCRIPTION
**Pull request by mcbridematt**
This is an effort to bring the kernel config in armsr/armv8 closer to the Linux arm64 defconfig, so armsr/armv8 can run out of the box on as many platforms as possible.

Changes of note:

- Enable virtualization/KVM host
- Sync CPU errata options set with Linux defconfig
- Sync CPU features and scheduler configuration with Linux defconfig
- Fix NXP i.MX8 support (was 'damaged' in the process of updating to kernel 6.1, 5.15/23.05 is unaffected)
- Enable Renesas SoCs (RZ/G, RZ/V)
- Add Raspberry Pi 4 network support
This is *not* a replacement for `bcm27xx/bcm2711`, as it relies on an external EFI firmware. But it is useful to have for testing, especially when using an ACPI firmware.
- Added Marvell (former Cavium IP) ThunderX2 network driver. I missed this previously and assumed ThunderX and ThunderX2 had the same driver, it's different
- Package and install `kmod-mdio-thunder`. This was already being built (loosely selected by ThunderX) but not installed

### CODE REVIEW

File: target/linux/armsr/armv8/config-6.1 [hunk 1]
```diff
@@ -22,33 +51,58 @@ CONFIG_ARCH_WANTS_NO_INSTR=y
...
+CONFIG_ARM64_ERRATUM_1542419=y
+CONFIG_ARM64_ERRATUM_1742098=y
+CONFIG_ARM64_ERRATUM_2051678=y
+CONFIG_ARM64_ERRATUM_2054223=y
+CONFIG_ARM64_ERRATUM_2067961=y
+CONFIG_ARM64_ERRATUM_2077057=y
+CONFIG_ARM64_ERRATUM_2441007=y
+CONFIG_ARM64_ERRATUM_2441009=y
+CONFIG_ARM64_ERRATUM_2457168=y
+CONFIG_ARM64_ERRATUM_2658417=y
```
Comment (hauke) [inline target/linux/armsr/armv8/config-6.1:77]: Please activate these too:
```
ARM64_ERRATUM_2253138
ARM64_ERRATUM_2224489
ARM64_ERRATUM_2054223
ARM64_ERRATUM_2067961
```

Comment (hauke) [reply target/linux/armsr/armv8/config-6.1:77]: It looks like these are not in the default configuration:
````
hauke@hauke-t480:~/linux/linux-next-clean$ git grep ARM64_ERRATUM_2253138
Documentation/arm64/silicon-errata.rst:| ARM            | Neoverse-N2     | #2253138        | ARM64_ERRATUM_2253138       |
arch/arm64/Kconfig:config ARM64_ERRATUM_2253138
arch/arm64/kernel/cpu_errata.c:#ifdef CONFIG_ARM64_ERRATUM_2253138
hauke@hauke-t480:~/linux/linux-next-clean$ git grep ARM64_ERRATUM_2224489
Documentation/arm64/silicon-errata.rst:| ARM            | Cortex-A710     | #2224489        | ARM64_ERRATUM_2224489       |
Documentation/arm64/silicon-errata.rst:| ARM            | Cortex-X2       | #2224489        | ARM64_ERRATUM_2224489       |
arch/arm64/Kconfig:config ARM64_ERRATUM_2224489
arch/arm64/kernel/cpu_errata.c:#ifdef CONFIG_ARM64_ERRATUM_2224489
hauke@hauke-t480:~/linux/linux-next-clean$ git grep ARM64_ERRATUM_2054223
Documentation/arm64/silicon-errata.rst:| ARM            | Cortex-A710     | #2054223        | ARM64_ERRATUM_2054223       |
arch/arm64/Kconfig:config ARM64_ERRATUM_2054223
arch/arm64/kernel/cpu_errata.c:#ifdef CONFIG_ARM64_ERRATUM_2054223
hauke@hauke-t480:~/linux/linux-next-clean$ git grep ARM64_ERRATUM_2067961
Documentation/arm64/silicon-errata.rst:| ARM            | Neoverse-N2     | #2067961        | ARM64_ERRATUM_2067961       |
arch/arm64/Kconfig:config ARM64_ERRATUM_2067961
arch/arm64/kernel/cpu_errata.c:#ifdef CONFIG_ARM64_ERRATUM_2067961
hauke@hauke-t480:~/linux/linux-next-clean$ 
````
They were recently added, maybe they are not so important.

Comment (mcbridematt) [reply target/linux/armsr/armv8/config-6.1:77]: > ARM64_ERRATUM_2253138
> ARM64_ERRATUM_2224489
> ARM64_ERRATUM_2054223
> ARM64_ERRATUM_2067961

These four are all relate to `CORESIGHT_TRBE` (CoreSight Trace Buffer extension) which does not get enabled on the default config or OpenWrt. On OpenWrt, CoreSight itself is disabled in the generic target config: https://github.com/openwrt/openwrt/blob/main/target/linux/generic/config-6.1#L1062

**Comment by mcbridematt** [comment]
A couple more changes added:

* Default `CONFIG_ARM_PMU=y` (Arm Performance Monitor Unit) to on for armsr/armv8. Without PMU enabled, it is not possible to launch guest VMs using KVM.
  The change has to be done inside `Config-kernel.in` as this appears to override any setting in the target config-X file.
* Enabled ACPI Button driver
* Enabled AHCI/SATA for relevant platforms (mvebu, layerscape, arm juno dev board)
* Enabled PCI(e)-related options that were not in the OpenWrt configuration
* Fixed a kernel configuration warning due to `CONFIG_FSL_ENETC_QOS` not being a kernel module.

For reference, this is the current diff between Linux 6.1.45 arm64 defconfig and the config generated after this change:

https://gist.github.com/mcbridematt/ec4a0ec782a650927aa1a897e3b29440

(The erratas that are currently turned off relate to 32-bit compatibility, e.g only appear if CONFIG_COMPAT=y)

**Pull request merged**